apps/dashboard/lib/trpc/routers/key/create.ts

```diff
@@ -135,3 +140,60 @@ export const createKey = rateLimitedProcedure(ratelimit.create)
 
     return { keyId, key };
   });
+
+async function upsertIdentity(
+  db: Database,
+  workspaceId: string,
+  externalId: string,
+): Promise<Identity> {
+  let identity = await db.query.identities.findFirst({
+    where: (table, { and, eq }) =>
+      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
+  });
+  if (identity) {
+    return identity;
+  }
+
```

Comment:
Why are you checking this and then returning it again. 

You shouldn't need to check the identity and then if it exists return it, it is already there.

Reply:
Sorry if I missed anything, but I thought we had to do an upsert here.  As mentioned in the issue, I also referred to the code [here](https://github.com/unkeyed/unkey/blob/main/apps/api/src/routes/v1_keys_createKey.ts#L574)

Reply:
Sorry for the delay my notifications didn't ping, if you wanted to check for upserting. Why not do and early return the other way. 

You current implementation just says query the db and set to variable identity, check if the identity exists and then return the identity.

```ts
 let identity = await db.query.identities.findFirst({
    where: (table, { and, eq }) =>
      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
  });
  if (identity) {
    return identity;
  }
```
When really could do:

```ts
 let identity = await db.query.identities.findFirst({
    where: (table, { and, eq }) =>
      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
  });
  if (!identity) {
   // run inserts
  }
```
or 
```ts
 let identity = await db.query.identities.findFirst({
    where: (table, { and, eq }) =>
      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
  });
 
 //handle the identity further down
```

Reply:
Got it! Understood what you meant. Thank you!